### PR TITLE
Feat: new plugin - vault : rename the default namespace with "hashicorp" and install vault by replicas: 3

### PR DESCRIPTION
**Pull request by KeHaohaoke**
Signed-off-by:  KeHao [hao.ke@merico.dev](mailto:hao.ke@merico.dev)

# Summary

Add harshicorp-vault plugin

## Key Points

- [ ] This is a breaking change.

- [x]  Documentation (new or existing) is updated.

## Description

 Downloading the all-in-one YAML file from hashicorp vault repo and applying with Helm. Install Vault on a existing k8s with replicas: 3
### Related Issues

close: #425 

### Screenshots

- **init** 

image

- **apply**
image

- **config.yaml**
image

- **devstream.state**
image

- **check status by kubectl after perform initialization according to  [Initialize and unseal Vault](https://learn.hashicorp.com/tutorials/vault/kubernetes-minikube?in=vault/kubernetes#initialize-and-unseal-vault)**
image

- **delete**
image

### CODE REVIEW

**Comment by KeHaohaoke** [comment]
doc for vault plugin : https://github.com/devstream-io/website/pull/13

**Comment by daniel-hutao** [comment]
#425 is a pr, not issue. And why do you created 2 pr for this feature?

**Comment by KeHaohaoke** [comment]
> #425 is a pr, not issue. And why did you create 2 pr for this feature?

#425 pr rom myself repo. This  pr  from merico-dev according to develop regulation.

**Comment by KeHaohaoke** [comment]
https://github.com/devstream-io/devstream/pull/435/commits/b0cf67fea0db165bd1d43480a70d26981a08ad57

modify name and namespace

**Comment by IronCore864** [comment]
> doc for vault plugin : [devstream-io/website#13](https://github.com/devstream-io/website/pull/13)

Doc is reviewed. The quality is not good. Please rework.

**Comment by IronCore864** [comment]
> #425 is a pr, not issue. And why did you create 2 pr for this feature?

@KeHaohaoke "Closes" should be used on "issues", not "PRs."

**Comment by IronCore864** [comment]
Duplicated commit messages, and one commit message failed lint-commit-message.

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

File: cmd/plugin/hashicorp-vault/main.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/devstream-io/devstream/pkg/util/log"
+)
+
+// NAME is the name of this DevStream plugin.
+const NAME = "hashicorp-vault"
+
+// Plugin is the type used by DevStream core. It's a string.
+type Plugin string
+
+// Create implements the create of vault.
```
[the remark this answers is not in the document]
Comment (KeHaohaoke) [reply cmd/plugin/hashicorp-vault/main.go]: > Comment error with plugin name

Corrected comments in source file

File: internal/pkg/plugin/hashicorpvault/create.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import (
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+
+	. "github.com/devstream-io/devstream/internal/pkg/plugin/common/helm"
+	"github.com/devstream-io/devstream/pkg/util/log"
+)
+
+// Create creates hashicorpvault with provided options.
```
[the remark this answers is not in the document]
Comment (KeHaohaoke) [reply internal/pkg/plugin/hashicorpvault/create.go]: > hashicorp-vault

Corrected comments in source file

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**